### ISSUE TITLE
[Tune] ray.tune.sklearn creates unfulfillable resource requests 

### ISSUE THREAD

**Issue by crclark** [ray-project/ray#28540]
### What happened + What you expected to happen

The code below hangs forever with an error from the scheduler: 

`(scheduler +3s) Error: No available node types can fulfill resource request {'CPU': 64.0}. Add suitable node types to this cluster to resolve this issue.`

The number of CPUs requested by Ray Tune varies depending on the size of the nodes in the cluster: bigger nodes => bigger, unfulfillable resource requests, so provisioning a larger cluster doesn't help.

### Versions / Dependencies

Ray 2.0

### Reproduction script

```
from ray.tune.sklearn import TuneGridSearchCV
import pandas as pd
from sklearn.dummy import DummyClassifier

df = pd.DataFrame({'x': [1,2,3], 'y': [0,1,1]})
search = TuneGridSearchCV(DummyClassifier(), {'strategy':['most_frequent', 'prior']}, n_jobs=1)
search.fit(df['x'], df['y'])
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.
